src/frameworks/next/utils.ts

```diff
@@ -421,3 +421,17 @@ export async function hasStaticAppNotFoundComponent(
 ): Promise<boolean> {
   return pathExists(join(sourceDir, distDir, "server", "app", "_not-found.html"));
 }
+
+/**
+ * Remove development files from the .next directory for production deployments.
+ */
+export async function removeDevFiles(distDirPath: string): Promise<void> {
+  const cacheDevFiles = join(distDirPath, "cache", "**", "*-development*");
+  const cacheEslintFiles = join(distDirPath, "cache", "eslint");
+
+  const filesToRemove = globSync(`{${cacheDevFiles},${cacheEslintFiles}}`, {
```

Comment:
alt, `[cacheDevFiles,cacheEslintFiles].join(",")`

why not use async here?

Reply:
I followed whatever's already used here, I'm hesitant to alter the glob usage per this: #6166. 

We have `node-glob` v7 which only supports async with a callback. If we bump it to latest we could use the `await` syntax without manually creating a Promise, can we do that?

Reply:
switched to the async glob, PTAL. However, AFAIK we can't create stubs of `glob` that way, so I couldn't write unit tests for it. Either way, added an E2E test which is better in this case.